### ISSUE TITLE
Feature Request - Add feature to enable power on before starting print

### ISSUE THREAD

**Issue by David-Maisonave** [jneilliii/OctoPrint-WemoSwitch#49]
Can a new option be added to enable power on before starting a print job?

I've been using this plugin for about 6 months now, and I've only figured out today that I'm able to have the printer turn on automatically before starting a print job by enabling the GCODE Trigger and adding the M80 command to the GCODE script [Before print job starts].

I think many users want this functionality, but don't know how to set it up.  IMHO, it would help these users if there was an option to do this programmatically.

Another option would be to add clear instructions next to the GCODE Trigger option on how to configure for automatic power on before starting a print job.

**Comment by jneilliii** [comment]
Unfortunately, this won't work for users that don't backpower their printer from USB connection from the Pi. In those instances there is no connection to the printer to send gcode commands so even the work around that you found won't work for those people. Also, in a disconnected state you cannot select a print to start, so that event monitoring is not possible either. There is however upload event monitoring that if you check the box to autostart your print when sending from PrusaSlicer/SuperSlicer/Slic3r it will auto power on the printer for that file and start the print. If you are using Cura I think there is integration in the OctoPrint Connection plugin (found in Cura Marketplace) that allows you to auto power on with upload as well.

**Comment by David-Maisonave** [comment]
I have Windows OctoPrint installed, and I believe the majority of Windows OctoPrint users connect to their printers via USB.
A disconnected state is not an issue, because by default the Wemo plugin connects after power is turned on.
That means the work around should work for most Windows OctoPrint installs.

No option works or is suitable 100% of the time.  That's why they're made as options.

Slicer's (including Cura) don't help, when printing from an already uploaded file.
With the workaround, it works with an already uploaded file, and with any slicer (without having to configure the slicer).  If a user has multiple slicers, they don't have to configure multiple applications.  The configuration is done in one place, and that means if the user changes the wemo switch or renames it, the user only has to fix it in one place.

I have the latest version of Cura installed, and the Cura-OctoPrint "Turn on printer" does **NOT** work.  
If the Wemo option "On on Upload" is **disabled**, and Cura-OctoPrint "Turn on printer" is **configured**, the printer does not turn on when starting a print job from Cura.
However, if Wemo option "On on Upload" is **enabled**, and Cura-OctoPrint "Turn on printer" is **de-configured**, the printer turns on when starting a print job from Cura.

Also if the **workaround is in place**, Wemo option "On on Upload" is **disabled**, and Cura-OctoPrint "Turn on printer" is **de-configured**, the printer turns on when starting a print job from Cura.

In this case, the workaround actually works better than the Cura-OctoPrint "Turn on printer" option.

**Comment by jneilliii** [comment]
> Also if the workaround is in place, Wemo option "On on Upload" is disabled, and Cura-OctoPrint "Turn on printer" is de-configured, the printer turns on when starting a print job from Cura.

This is interesting to me. I didn't even catch the fact that your workaround bypasses the printer being disconnected? I didn't think you could start a print in that state, and therefore a beforePrintStart gcode wouldn't be possible either.

> I have the latest version of Cura installed

I haven't been able to verify this version yet (assuming 4.9), I installed but it's flaking out big time and I think it may be plugin related.

**Comment by David-Maisonave** [comment]
I have Cura 4.9 installed.
The OctoPrint GUI doesn't let you start a print job when disconnected, but surprisingly you can start a print job via Cura-OctoPrint while OctoPrint is disconnected.
It would be nice if a plugin was created with option to let OctoPrint start the print job in a disconnected state.

FYI:
There is a minor issue I'm seeing with the Wemo plugin's power timeout.  I see the timeout getting displayed a second time before the previous timeout has closed or timed out.  I think it only happens when the timeout value is longer than the idle timeout.
Do you want me to create a new issue with more details?

**Comment by jneilliii** [comment]
> Do you want me to create a new issue with more details?

Yes please, that's more a bug than feature enhancement.

**Comment by David-Maisonave** [comment]
Work around solves this issue. See #57
